src/argilla/client/client.py

```diff
@@ -137,6 +138,7 @@ def __init__(
                 " datasets, specify a valid workspace name on `rg.init` or set it"
                 " up through the `rg.set_workspace` function.",
                 category=UserWarning,
+                stacklevel=2,
```

Comment:
I'm not a huge fan of `stacklevel=2` for every warning, as in some cases we just want to display the warning itself, not any additional lines, so this tends to make `warnings` longer for not a clear reason in most of the cases, shouldn't we carefully evaluate this one before including it as a default?

Reply:
As of https://docs.python.org/3/library/warnings.html#warnings.warn I understand this is useful when we wrap up the `warnings` inside a function, but this is not the case right?

Reply:
Yes I know, but sometimes the line where the `warnings` is raised is not relevant e.g.

```python
if deprecated_arg is not None:
  warnings.warn("`deprecated_arg` is deprecated", DeprecationWarning, stacklevel=2)
```

Here the user will see the line `if deprecated_arg is not None`, which doesn't add any relevance and/or context to the `warnings` message as it's pretty self explanatory, this is why I meant we should revisit this carefully to add `stacklevel=2` just when relevant 😄

Reply:
Anyway I'm still open to discussion in case you guys think that setting `stacklevel=2` as default makes more sense in most of the cases

Reply:
I think the default is showing the `warnings.warn` line, which is kind of useless. In your example, I would rather see e.g. 

```
DeprecationWarning: `deprecated_arg` is deprecated
   output = a_func_with_warning(deprecated_arg) in demo.py#7
```
or however it formats it exactly. Regardless - that it shows where the user made the mistake.
But then we indeed need to go case-by-case to see where we need `stacklevel=2` or where we need e.g `stacklevel=3`.

Thoughts?

Reply:
Sure, then we can talk tomorrow and decide whether to go ahead with `stacklevel=2` or the default for the moment, and then we can start using `stacklevel=3` when needed